test/browser/spec/ol/source/ImageStatic.test.js

```diff
@@ -60,31 +69,49 @@ describe('ol/source/ImageStatic', function () {
       const image = source.getImage(extent, resolution, pixelRatio, projection);
 
       source.on('imageloadend', function (event) {
-        expect(image.getImage().width).to.be(512);
-        expect(image.getImage().height).to.be(256);
-        done();
+        try {
+          const [resolutionX, resolutionY] = image.getResolution();
+          expect(resolutionX).to.be(2 * resolutionY);
+          done();
+        } catch (e) {
+          done(e);
+        }
       });
 
       image.load();
     });
 
-    it('respects imageSize', function (done) {
+    it('Had an imageSize option which changed the imageExtent when used with incorrect image sizes', function (done) {
```

Comment:
Keeping this test for historic reasons, to show that what `imageSize` did was neither useful nor what was advertised.